src/classes/BDI_DataImportService.cls

```diff
@@ -771,89 +773,31 @@ global with sharing class BDI_DataImportService {
-                    //in a non-npsp namespace
-                    if (UTIL_Namespace.getNamespace() != 'npsp') {
-                        String fieldNameOld = fieldName;
-                        fieldName = UTIL_Namespace.StrTokenRemoveNSPrefix(fieldName);
-
-                        //if we have a detected namespace, and removed a namespace from the current field
-                        //add the current namespace back in, allowing this code to run in any namespace
-                        if (fieldName != fieldNameOld && UTIL_Namespace.getNamespace() != '') {
-                            fieldName = UTIL_Namespace.StrAllNSPrefix(fieldName);
-                        }
-                    }
-
-                    // make sure they have permissions to modify the target field
-                    // we can't do this in all tests, because many profiles won't have FLS set for all our new fields!
-                    if (!Test.isRunningTest() || failFieldLevelSecurity) {
-                        fieldDescribe = UTIL_Describe.getFieldDescribe(targetObjectName, fieldName);
-                        Set<String> setAuditFields = new Set<String>{
-                            'createddate', 'createdbyid', 'lastmodifieddate', 'lastmodifiedbyid'
-                        };
-                        Boolean isAuditField = setAuditFields.contains(fieldName.toLowerCase());
-
-                        if (failFieldLevelSecurity ||
-                            fieldDescribe == null ||
-                            !fieldDescribe.isCreateable() ||
-                            (!isAuditField && !fieldDescribe.isUpdateable())) {
-                                throw (new FLSException(
-                                    String.format(System.Label.flsError, new List<String>{ targetObjectName + '.' + fieldName }
-                                )));
-                        }
-                    }
-
-                    dataImportFieldToTargetField.put(dataImportField, fieldName);
-                }
-            }
-        }
-
-        if (isCurrencyMapped(targetObjectName)) {
-            addCurrencyIsoCode(dataImportFieldToTargetField);
-        }
-
-        return dataImportFieldToTargetField;
-    }
-
-    /*******************************************************************************************************
-    * @description Checks if this is one of the objects where we want to have a currency mapped.
-    * As of now, that object is Opportunity. Payment CurrencyISOCode is inherited from the Opportunity.
-    * @param strBaseObj the underlying object the bdi object is (ie, Contact, Account, etc)
-    * @return Boolean if the strBaseObj is an Opportunity.
-    */
-    private static Boolean isCurrencyMapped(String objectName) {
-        return objectName.equalsIgnoreCase('Opportunity');
+    global static Map<String, String> mapFieldsForDIObject(String dataImportObjectName, 
```

Comment:
I think this method is still being used in BDI_MatchDonations. Would that be an issue if this now always only returns an empty map?

Reply:
Yeah, I think one of two things needs to be done:  1) once the custom setting field is in place to control mapping we could modify this to automatically construct the right version the Data import service based on the setting.  

Or 2)We need to remove all references to this method in our code and throw an error if it is called.

Reply:
Good find - I'm in favor of pulling all our internal references to the method for SURE, and I updated this PR to (I think) take out all the remaining calls. I do think it might be wise to support this method for the short term just in case a customer is calling it, but we should probably add the  "Deprecated" annotation, which I believe is designed for this use case  - https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_deprecated.htm